### ISSUE TITLE
Example of `InputChip`s  being generated from user's input.

### ISSUE THREAD

**Issue by TahaTesser** [flutter/flutter#128247]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

[Material guidelines ](https://m3.material.io/components/chips/guidelines) show `InputChip` being generated by user inputs.

![image](https://github.com/flutter/flutter/assets/48603081/db4be25d-c298-4325-b380-3311328a1cd2)

https://github.com/flutter/flutter/assets/48603081/cd122a9a-cf47-4c7d-b79d-baa23c84b729

Current documentation doesn't showcase how to achieve this. 
https://master-api.flutter.dev/flutter/material/InputChip-class.html

I found an old but really cool example by @slightfoot 

https://stackoverflow.com/a/52415307

### Proposal

Add a useful example where `InputChip`s are generated by user inputs.

**Comment by slightfoot** [comment]
Wow that was a long time ago 😅

**Comment by TahaTesser** [comment]
cc: @HansMuller

**Comment by HansMuller** [comment]
An updated version of this example, or something similar would be really useful. This is an important use case.  The original example (https://gist.github.com/slightfoot/c6c0f1f1baca326a389a9aec47886ad6) is copyrighted, so I think we'd need something new but with similar capabilities.

**Comment by TahaTesser** [comment]
I'll come back to this in a couple of weeks.

(Anyone want to use fix this before, feel free to file PR)

**Comment by SajaHerez** [comment]
I working on  eample of InputChips being generated from user's input.
How I  can  do   PR ??

**Comment by HansMuller** [comment]
https://www.youtube.com/watch?v=4yBgOBAOx_A

**Comment by andreacioni** [comment]
Hello everyone,

I've just opened a PR to try to solve this issue. I've studied the subject using the sample by @slightfoot (https://gist.github.com/slightfoot/c6c0f1f1baca326a389a9aec47886ad6) and then I heavily changed it to be Flutter 3 compatible and more easy to understand (or at least this is what I had in mind 😉). 

I'm not sure about how copyright works in this case and if my adaptation could be considered safe to be merged.
